### ISSUE TITLE
upgrade.php and create.sql different

### ISSUE THREAD

**Issue by shawniverson** [mailwatch/1.2.0#681]
It appears that the initial setup of the databases with create.sql are not the same as those resulting from running upgrade.php.

Should create.sql be brought up to the same level as upgrade.php?

create.sql
```
CREATE TABLE IF NOT EXISTS `users` (
  `username` varchar(191) COLLATE utf8_unicode_ci NOT NULL DEFAULT '',
  `password` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL,
  `fullname` varchar(255) COLLATE utf8_unicode_ci NOT NULL DEFAULT '',
  `type` enum('A','D','U','R','H') COLLATE utf8_unicode_ci DEFAULT NULL,
  `quarantine_report` tinyint(1) DEFAULT '0',
  `spamscore` float DEFAULT '0',
  `highspamscore` float DEFAULT '0',
  `noscan` tinyint(1) DEFAULT '0',
  `quarantine_rcpt` varchar(60) COLLATE utf8_unicode_ci DEFAULT NULL,
  `resetid` varchar(32) COLLATE utf8_unicode_ci DEFAULT NULL,
  `resetexpire` bigint(20) COLLATE utf8_unicode_ci DEFAULT NULL,
  `lastreset` bigint(20) COLLATE utf8_unicode_ci DEFAULT NULL,
  PRIMARY KEY (`username`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
```

upgrade.php result
```
| users | CREATE TABLE `users` (
  `username` varchar(191) COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',
  `password` varchar(255) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
  `fullname` varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',
  `type` enum('A','D','U','R','H') COLLATE utf8mb4_unicode_ci DEFAULT NULL,
  `quarantine_report` tinyint(1) DEFAULT '0',
  `spamscore` float DEFAULT '0',
  `highspamscore` float DEFAULT '0',
  `noscan` tinyint(1) DEFAULT '0',
  `quarantine_rcpt` varchar(60) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
  `resetid` varchar(32) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
  `resetexpire` bigint(20) DEFAULT NULL,
  `lastreset` bigint(20) DEFAULT NULL,
  PRIMARY KEY (`username`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci |
```

**Comment by endelwar** [comment]
`create.sql` is more compatible with older mysql installation with no utf8mb4 support.
`upgrade.php` enhance the charset and collation on capable mysql systems.
In install procedure there should be a step to run upgrade.php after create.sql

**Comment by shawniverson** [comment]
Ah, I see.  That wasn't clear to me.  Makes a lot more sense now.

**Comment by Skywalker-11** [comment]
I think its not realy intuitive that you have to run upgrade.php on a fresh install to be able to use utf8mb4. 
At least we should mention it in the docs

**Comment by remkolodder** [comment]
I agree with @Skywalker-11 . If you do an initial install, it should be the best experience you can get. Everyone expects to run a 'creation' tool so that the proper database structure is created. You would not expect to instantly upgrade (that suggests upgrade.php) the structure to get some additional features. If the database supports utf8mb4, then we should create those tables. Else it should fallback to regular utf8.

**Comment by stefaweb** [comment]
A suggestion.

You can replace create.sql by a small install_database.php script.

You test if utf8mb4 is possible and load the right database version. 
SQL schema included in the php. No external file.

**Comment by remkolodder** [comment]
Or via a general install.php file which writes conf.php and installs the db etc.

**Comment by stefaweb** [comment]
> Or via a general install.php file which writes conf.php and installs the db etc.

@Skywalker-11 is working on this but its not yet ready.

https://github.com/mailwatch/mailwatch-install-script

**Comment by Skywalker-11** [comment]
We need some people to test the script on different OS. Especially sendmail as MTA is still missing. Heres a list what we have to do : mailwatch/mailwatch-install-script#2

I only have experience with debian/ubuntu and postfix 😞

**Comment by stefaweb** [comment]
And this one?

**Comment by shawniverson** [comment]
I think this one should remain open for further consideration.   I think the create.sql and upgrade.php logic could be combined into a unified database preparation script.  Not really necessary right now, but perhaps down the road.

**Comment by asuweb** [comment]
The simple fix is for upgrade.php to run create.sql if the database doens't exist or is empty.

**Comment by stefaweb** [comment]
Does this still active or can be closed ?

**Comment by Skywalker-11** [comment]
Still needs to be done
